packages/shared/src/components/sources/SourceActions/SourceActionsFollow.tsx

```diff
@@ -12,15 +13,32 @@ interface SourceActionsFollowProps {
 
 const SourceActionsFollow = (props: SourceActionsFollowProps): ReactElement => {
   const { className, isSubscribed, isFetching, onClick, variant } = props;
-  const label = isSubscribed ? 'Unfollow' : 'Follow';
+  const [isHovered, setIsHovered] = useState(false);
+
+  let label = isSubscribed ? 'Following' : 'Follow';
+
+  if (isHovered && isSubscribed) {
+    label = 'Unfollow';
+  }
```

Comment:
Yeah I tried handling it with CSS, I gave `after:content` a try but to no luck. 

Above is ok but kinda feels wrong to render two divs all the time?

Reply:
I like it since its more performant then JS for sure though.

Reply:
Two div are present but only one is rendered if we use display:none, so this is a ok situation of accessibility (display none is not read nor rendered by any device) 👀 

I suggest maybe to use a neutral sentence for `aria-label`, something like this `Toggle follow status, currently you are {subscribed/x}`.

Most of the people that use aria attributes are using a screen reader that focus on element more then hover so this could be a good trade off. 

What do you think?

Reply:
Let's do it

Reply:
updated!